### PR TITLE
feat: pre-calculate decoders in signature model

### PR DESCRIPTION
**Pull request by Goldziher**
This PR improves our deserialization by pre-calculating the required decoders for the signature. 

I added a hack to bypass msgspec, which strips Annotated before hitting the hooks. I hope this is temporary - see: https://github.com/jcrist/msgspec/issues/497

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -23,14 +23,9 @@ documentation = "https://docs.litestar.dev/"
 keywords = [
     "api",
     "rest",
-    "http",
```
Comment (cofin) [inline pyproject.toml:26]: What's the reason for removing some of these keywords?

File: litestar/_signature/types.py [hunk 2]
```diff
@@ -9,3 +9,7 @@ class ExtendedMsgSpecValidationError(ValidationError):
     def __init__(self, errors: list[dict[str, Any]]) -> None:
         self.errors = errors
         super().__init__(errors)
+
+
+class SerializationWrapper:
```
Comment (provinzkraut) [inline litestar/_signature/types.py:14]: This doesn't seem to get used anywhere? Did you accidentally drop some commits along the way? :upside_down_face:

File: pyproject.toml (hunk 1 above)
Comment (Goldziher) [reply pyproject.toml:26]: linting errors in my IDE letting me know that keywords are limited to 5 max. intellij got updated.

File: litestar/_signature/types.py (hunk 2 above)
Comment (Goldziher) [reply litestar/_signature/types.py:14]: ha yes, lemme remove it.

**Review by provinzkraut — APPROVED** [review]

**Pull request merged**